### PR TITLE
Initial table of contents

### LINKED ISSUE #13: Create the Table Of Contents for the new guide
Create the Table Of Contents for the new guide outlining the steps we wish to feature for microprofile-config.

### PR DESCRIPTION
**Pull request by dmorgant**
fixes #13

### CODE REVIEW

File: js/interactive-guides/microprofile-config/microprofile-config-callback.js [hunk 1]
```diff
@@ -10,7 +10,91 @@
 *******************************************************************************/
 var microprofileConfigCallBack = (function() {
 
+<<<<<<< HEAD
```
Comment (lavenal) [inline js/interactive-guides/microprofile-config/microprofile-config-callback.js]: Hmm, why do we have the merge lines here? Have Steven moved the codes to guides-common?

File: json-guides/microprofile-config.json [hunk 2]
```diff
@@ -12,58 +12,129 @@
...
+                "Begin by enabling the MicroProfile Config feature in your <code>server.xml</code> file."
             ],
             "instruction": [
-                "Add the <code>@Inject @ConfigProperty(name=\"download-url\", defaultValue=\"ftp://music.com/us/download\")</code> on line 9, after the <code>private Config config</code> or click <action title='Configure inject using ConfigProperty' onclick=\"microprofileConfigCallBack.addConfigInjectButton(event)\"><b>@Inject @ConfigProperty(name=\"download-url\", defaultValue=\"ftp://music.com/us/download\")</b></action>. Then, click <action title='Save' onclick=\"microprofileConfigCallBack.saveButton(event)\"><b>Save</b></action> on the editor menu pane."
+              "Add <b>&lt;feature>mpConfig-1.1&lt;/feature>."
+            ]
+        },
+        {
+            "name": "ConfigureViaInject",
+            "title": "Injecting config properties through annotations",
```
Comment (lavenal) [inline json-guides/microprofile-config.json]: I think it is better to use "Configuring configuration properties through injections" but I could be wrong.

File: js/interactive-guides/microprofile-config/microprofile-config-callback.js (hunk 1 above)
Comment (dmorgant) [reply js/interactive-guides/microprofile-config/microprofile-config-callback.js]: Dang!   I'm not sure why those lines are there either.  I had no indication that there was a merge conflict when I the pushed.   I will remove the lines as was done in Steven's work item.

File: json-guides/microprofile-config.json (hunk 2 above)
Comment (dmorgant) [reply json-guides/microprofile-config.json]: I'll change to 'Injecting configuration properties through annotations' and we will discuss more as a team.

File: js/interactive-guides/microprofile-config/microprofile-config-callback.js (hunk 1 above)
Comment (dmorgant) [reply js/interactive-guides/microprofile-config/microprofile-config-callback.js]: Dang!   I'm not sure why I was not notified of the merge conflict!.....I'm removing the lines as Steven had done.

File: json-guides/microprofile-config.json [hunk 3]
```diff
@@ -12,58 +12,129 @@
...
+                "Injecting a configuration value this way is recommended for mandatory properties. If no configured value exists for this property, a <code>DeploymentException</code> will be thrown during startup."
             ],
+            "instruction": [
+                "To inject a <code>download-url</code> configuration property into the code, add the following <code>@Inject</code> and <code>@ConfigProperty</code> annotations to line 6 before declaring the <code>String download-url</code>, or click  <action title='Inject a ConfigProperty'><b>@Inject @ConfigProperty(name=\"download-url\")</b></action>. Then, click <action title='Save'><b>Save</b></action> on the editor menu pane.<br><br><note>Note: You have not yet specified a value for the <code>download-url</code> configuration property.  Execution of this code at this time would cause a <code>DeploymentException</code> to be thrown since a value is required.</note>"
```
Comment (lavenal) [inline json-guides/microprofile-config.json]: There is no callback to inject the line of code into the editor. Please open an issue to complete this step. We also need to decide whether to use a browser to get the DeploymentException or not.

Comment (dmorgant) [reply json-guides/microprofile-config.json]: Issue #53 opened.

**Pull request merged**